lib/index.js

```diff
@@ -31,18 +32,26 @@ VM.deps = {
  * @param {StateManager} [opts.stateManager] A state manager instance (EXPERIMENTAL - unstable API)
  * @param {Trie} [opts.state] A merkle-patricia-tree instance for the state tree (ignored if stateManager is passed)
  * @param {Blockchain} [opts.blockchain] A blockchain object for storing/retrieving blocks (ignored if stateManager is passed)
+ * @param {String|Number} opts.chain The chain the VM operates on [default: 'mainnet']
+ * @param {String} opts.hardfork Hardfork rules to be used [default: 'byzantium', supported: 'byzantium']
  * @param {Boolean} [opts.activatePrecompiles] Create entries in the state tree for the precompiled contracts
  * @param {Boolean} [opts.allowUnlimitedContractSize] Allows unlimited contract sizes while debugging (default: false; ONLY use during debugging)
  */
 function VM (opts = {}) {
   this.opts = opts
 
+  let chain = opts.chain ? opts.chain : 'mainnet'
+  let hardfork = opts.hardfork ? opts.hardfork : 'byzantium'
+  let supportedHardforks = [ 'byzantium' ]
+  this._common = new Common(chain, hardfork, supportedHardforks)
```

Comment:
IMO this should just be a static map from hard fork -> costs.  It doesn't seem to make sense to instantiate it as an object and then attach it to the current runState since it is external to the call state and static for the lifetime of a transaction.

Reply:
This is deeply rooted in the new [ethereumjs-common](https://github.com/ethereumjs/ethereumjs-common) library and there are various reasons for why this is designed to be created as an object, please have a closer look into the library if you want to understand what it does and the reasons for the design choices.

Reply:
One main thing it does is to encapsulate the ``hardfork`` choice switch, otherwise you would have to do in the using library (here: the VM) every time you want to access an parameter value, like:

```javascript
if (vm.hardfork === 'byzantium')  {
  // read param from byzantium.js file
} if (vm.hardfork === 'constantinople') {
  // ...
}
```

With instantiating the Common library with network and the hardfork, this is further done internally and the library makes the correct choice with:

```javascript
c.param('pow', 'minerReward')
```

Reply:
> One main thing it does is to encapsulate the hardfork choice switch, otherwise you would have to do in the using library (here: the VM) every time you want to access an parameter value, like:
```
if (vm.hardfork === 'byzantium')  {
  // read param from byzantium.js file
} if (vm.hardfork === 'constantinople') {
  // ...
}
```

Or you have all HF parameters in one file with the JSON structure that I pasted up above.  Then you don't need to do any if statement like this.

But... I digress.  It would be great to have the common parameters as a global const variable.  Not attached to `runState`.  Then I think this can be merged.

Reply:
What you are proposing is actually how it is done in the ``ethereumjs-common`` library (with the slight modification of not having this in dict but separate files like [byzantium.js](https://github.com/ethereumjs/ethereumjs-common/blob/master/hardforks/byzantium.json) or [spuriousDragon.js](https://github.com/ethereumjs/ethereumjs-common/blob/master/hardforks/spuriousDragon.json).

If you re-introduce this here, you loose all the encapsulation functionality of the library. The idea is to actually set the hardfork once (or in the future it will also be possible to have this determined by the block number, used for the client).

The ``c.param('pow', 'minerReward')`` reads are then automatically taking the correct value from the hardfork chosen (relevant if a value is overriden by a hardfork e.g.).

Since this whole library is designed in a stateful way, this has to be attached to the runState, otherwise the common class doesn't know the network/HF conditions it's operating in.

Reply:
If this has `hardfork` and `suppertedHardforks` as an argument, I'd assume it returns nil if `hardfork` is not part of the supported ones. Should this code here check if the call failed or returned nil meaning an unsupported fork was chosen?

Reply:
Answer is here:
> This will e.g. throw an error when a param is requested for an unsupported hardfork and like this prevents unpredicted behaviour.

Though perhaps a comment here may be good.

Reply:
@axic Added the comment you suggested to the JSDoc and README. Do you also want to have a broader look on this? I would otherwise merge within 24h as announced above.